### PR TITLE
Add Plotter class to plot_util module

### PR DESCRIPTION
**Pull request by yuyizheng1112**
This PR aims to solve the code overlap between  `SDECPlotter` and `InteractionRadiusPlotter`. Here I add `Plotter` class to `plot_util.py` as a composition of plotter tools, which encapsulates some common functions used by both plotters.

**Motivation and context**

Cope with code overlap in pr #1606 

**How has this been tested?**
- [ ] Testing pipeline.
- [x] Other. 

**Type of change**

- [ ] Bug fix. 
- [x] New feature. 
- [ ] Breaking change. 
- [ ] None of the above. 

**Checklist**

- [ ] My change requires a change to the documentation.
    - [ ] I have updated the documentation accordingly.
    - [ ] (optional) I have built the documentation on my fork following [the instructions](https://tardis-sn.github.io/tardis/development/documentation_preview.html).
- [x] I have assigned and requested two reviewers for this pull request.

### CODE REVIEW

**Review by jamesgillanders — APPROVED** [review]
Seems like a good idea to prevent duplication of code. Is this definitely where we want to put it? @MarkMageeAstro @marxwillia - if so, I'd be happy to merge, pending a second review

**Comment by MarkMageeAstro** [comment]
@yuyizheng1112, can you provide more information on the tests? Does making this change still behave as expected? I haven't run the code myself.

@jamesgillanders, I suppose it would go here or in the analysis folder. Since this doesn't really return anything, but is just used for plotting functions, it's probably fine to put here.

**Comment by andrewfullard** [comment]
This needs rebasing

File: tardis/visualization/plot_util.py
Comment (wkerzendorf) [inline tardis/visualization/plot_util.py:138]: this is a dangerous assumption IMHO

File: tardis/visualization/plot_util.py
Comment (wkerzendorf) [inline tardis/visualization/plot_util.py:145]: why `* 100`

Comment (jamesgillanders) [reply tardis/visualization/plot_util.py:145]: so that the IDs are all 3 or 4 digit numbers - e.g. Ca II = 2001

File: tardis/visualization/plot_util.py
Comment (jamesgillanders) [reply tardis/visualization/plot_util.py:138]: This should be fine as it is well documented that if the user doesn't include a space, the code will fall over. And it was the easiest way we could think of at the time for differentiating between a mixed list of ions and elements in a list.

File: tardis/visualization/plot_util.py
Comment (jamesgillanders) [reply tardis/visualization/plot_util.py:145]: FYI, these are all in the current implementation of the SDEC plotter - all this PR does is move these functions to a separate file to prevent duplication.

**Review by jaladh-singhal — COMMENTED** [review]
I think it's better to do these changes within #1606 because it will be easier to see why  so many of the methods have been moved? 

Also we need to consider if moving these methods don't make writing tests for SDEC plot even harder. And how have you tested if it doesn't break SDEC plot by running the notebook?

**Comment by yuyizheng1112** [comment]
@jaladh-singhal I think it makes sense to do this in another PR because #1606 and this PR are editing different modules.

For the second question, do you mean if it is harder to write tests for SDEC Plot together with `Plotter` class? 
And I ran the SDEC plot in the notebook. The plot works as usual.

**Comment by jaladh-singhal** [comment]
> For the second question, do you mean if it is harder to write tests for SDEC Plot together with Plotter class?

Not exactly harder but we might need to change the way we're planning to write tests now.

**Comment by andrewfullard** [comment]
Can you fix the merge conflict @yuyizheng1112 ? Then we can merge.

**Comment by wkerzendorf** [comment]
See #1838 as it pertains to this issue.

**Comment by wkerzendorf** [comment]
@visualization team - what is going on with this?

**Comment by jamesgillanders** [comment]
Closing this for now. The idea is sound, but we need to update the InteractionRadiusPlotter tool first

**Pull request closed without merging**
